src/invidious/videos.cr

```diff
@@ -877,42 +877,84 @@ def extract_video_info(video_id : String, proxy_region : String? = nil, context_
         }
   ).try { |a| JSON::Any.new(a) } || JSON::Any.new([] of JSON::Any)
 
-  primary_results = player_response.try &.["contents"]?.try &.["twoColumnWatchNextResults"]?.try &.["results"]?
-    .try &.["results"]?.try &.["contents"]?
-  sentiment_bar = primary_results.try &.as_a.select(&.["videoPrimaryInfoRenderer"]?)[0]?
-    .try &.["videoPrimaryInfoRenderer"]?
-      .try &.["sentimentBar"]?
-        .try &.["sentimentBarRenderer"]?
-          .try &.["tooltip"]?
-            .try &.as_s
-
-  likes, dislikes = sentiment_bar.try &.split(" / ", 2).map &.gsub(/\D/, "").to_i64 || {0_i64, 0_i64}
-  params["likes"] = JSON::Any.new(likes)
-  params["dislikes"] = JSON::Any.new(dislikes)
-
-  params["descriptionHtml"] = JSON::Any.new(primary_results.try &.as_a.select(&.["videoSecondaryInfoRenderer"]?)[0]?
-    .try &.["videoSecondaryInfoRenderer"]?.try &.["description"]?.try &.["runs"]?
-      .try &.as_a.try { |t| content_to_comment_html(t).gsub("\n", "<br/>") } || "<p></p>")
-
-  metadata = primary_results.try &.as_a.select(&.["videoSecondaryInfoRenderer"]?)[0]?
-    .try &.["videoSecondaryInfoRenderer"]?
-      .try &.["metadataRowContainer"]?
-        .try &.["metadataRowContainerRenderer"]?
-          .try &.["rows"]?
-            .try &.as_a
+  # Top level elements
+
+  primary_results = player_response
+    .dig?("contents", "twoColumnWatchNextResults", "results", "results", "contents")
+
+  video_primary_renderer = primary_results
+    .try &.as_a.find(&.["videoPrimaryInfoRenderer"]?)
+      .try &.["videoPrimaryInfoRenderer"]
```

Comment:
Bunch of lines in here don't have the safe access `#[]?` but should they?

Reply:
I'm not sure to understand what you mean. We try to cast `primary_results` to an array and `.find` a sub-element that has the `videoPrimaryInfoRenderer` key. Then we get to that element.
Though, yes, it's assumed that if the cast to `Array` is sucessful, it means that a `JSON::Any` element follows.